modules.d/90dmsquash-live/dmsquash-live-root.sh

```diff
@@ -33,19 +33,46 @@ overlay_size=$(getarg rd.live.overlay.size=)
 getargbool 0 rd.live.overlay.thin && thin_snapshot="yes"
 getargbool 0 rd.live.overlay.overlayfs && overlayfs="yes"
 
+# Take a path to a disk label and return the parent disk if it is a partition
+# Otherwise returns the original path
+function get_check_dev() {
+    local _udevinfo
+    dev_path="$(udevadm info -q path --name "$1")"
+    _udevinfo="$(udevadm info -q property --path "${dev_path}")"
+    strstr "$_udevinfo" "DEVTYPE=partition" || {
+        echo "$1"
+        return
+    }
+    parent="${dev_path%/*}"
+    _udevinfo="$(udevadm info -q property --path "${parent}")"
+    strstr "$_udevinfo" "DEVTYPE=disk" || {
+        echo "$1"
+        return
+    }
+    strstr "$_udevinfo" "ID_FS_TYPE=iso9660" || {
+        echo "$1"
+        return
+    }
+
+    # Return the name of the parent disk device
+    echo "$_udevinfo" | grep "DEVNAME=" | sed 's/DEVNAME=//'
+}
+
+# Find the right device to run check on
+check_dev=$(get_check_dev "$livedev")
```

Comment:
How about 
check_dev=$(lsblk -npo PKNAME "$livedev")

Reply:
Hm, yeah, that looks like it works for my test cases, but it doesn't have any kind of sanity checking for iso9660 on the parent device. I'm trying to limit the impact of the change as much as possible :)

Reply:
@bcl Don't lines 37-40 already test for iso9660 or udf filesystem types?

Reply:
Yes, but they don't fall back to livedev if they fail. In practice this likely doesn't matter, but I wanted the failure case to act as much like it does currently as possible.